Fix #2531

naoyam opened the pull request:
Changed ReplayTransformations to take bool parameters explicitly with set methods. This is to avoid accidentally passing those bool arguments in a wrong order. More verbose, but would be safer

Fixes #2531

Commit:
Fix #2531

Changed ReplayTransformations to take bool parameters explicitly with
set methods. This is to avoid accidentally passing those bool arguments
in a wrong order. More verbose, but would be safer

zasdfgbnm commented on third_party/nvfuser/csrc/transform_iter.cpp:
```diff
@@ -174,13 +174,8 @@ void ReplayTransformations::handle(Swizzle2D* swizzle_2d) {
 
 ReplayTransformations::ReplayTransformations(
     const std::vector<IterDomain*>& _target_domain,
```
Unrelated question: why are their names starting with `_`? Is this some naming convention?

naoyam replied:
I guess that's written by @csarofeen long time ago. Will change.

zasdfgbnm reviewed (approved):
LGTM, thanks for the clean up!

Commit:
cleanup

naoyam commented:
@zasdfgbnm I added one more commit (34a79b9), which doesn't change any logic but I just tried to clean the code up. Can you please review it as well?

naoyam commented on third_party/nvfuser/csrc/transform_iter.h:
```diff
@@ -39,80 +39,98 @@ struct id_int_lt {
 // operations it can't.
 class TORCH_CUDA_CU_API ReplayTransformations : public IterVisitor {
- protected:
-  // NOLINTNEXTLINE(cppcoreguidelines-non-private-member-variables-in-classes)
```
I didn't like this `NOLINTNEXTLINE` annotation, so I moved them to `private` when easily done. Still some have the annotation, but looks cleaner than before.

zasdfgbnm commented on third_party/nvfuser/csrc/transform_iter.h:
```diff
@@ -39,80 +39,98 @@ struct id_int_lt {
+  // Currently this options should always be off but
+  //  later we may have cases in scheduling large fusions where
+  //  this functionality could be useful.
+  // NOLINTNEXTLINE(cppcoreguidelines-non-private-member-variables-in-classes)
```
Is this still needed? It is private now.

naoyam replied:
Deleted

Commit:
Fix: do error check just before runReplay

Was done in the ctor, but at that point error_on_failue is always true,
which is not the behavior we want in some cases

Commit:
cleanup

naoyam commented:
There's one bug, which is fixed in [7c320ee](https://github.com/csarofeen/pytorch/pull/2532/commits/7c320eed8cac28d4899576400142e992ed187aec)

zasdfgbnm commented:
> 

What is the bug? The code looks equivalent to me, it just move the error checking from ctor to `runReplay`.

zasdfgbnm commented:
> > 
> 
> What is the bug? The code looks equivalent to me, it just move the error checking from ctor to `runReplay`.

Oh I see, in ctor, the `error_on_failure_` is not set.

naoyam commented:
Thanks. I'll merge it once all tests finish.

naoyam commented:
All are green. Merging.
